### PR TITLE
Test prefetching

### PR DESCRIPTION
**Pull request by dkonieczek**

### CODE REVIEW

**Comment by dkonieczek** [comment]
Not included in this PR:
- Disable lazy loading for first 4 Result components in Results.tsx
- Prefetching first 4 images:
```
cntrlr.on('afterStore', async ({ controller }: { controller: AbstractController }, next) => {
	// @ts-ignore - store type
	controller.store.results
		.slice(0, 4)
		.map((result: any) => result.mappings.core.imageUrl)
		.forEach((url: any) => {
			console.log('preloading image', url);
			const img = new Image();
			img.src = url;
		});

	await next();
});
```

File: packages/snap-preact/src/Snap.tsx [hunk 1]
```diff
@@ -55,9 +55,20 @@ export type ExtendedTarget = Target & {
...
+	url?: UrlTranslatorConfig & {
+		initial?: InitialUrlConfig;
+	};
+	context?: ContextVariables;
+};
+
 export type SnapConfig = {
 	features?: SnapFeatures;
 	mode?: keyof typeof AppMode | AppMode;
+	pageType?: 'search' | 'category';
```
Comment (korgon) [inline packages/snap-preact/src/Snap.tsx:71]: I'm not sure this should be configurable here - the current change is around prefetching, which they could set themselves in the targeter config. I'm thinking we remove this...

In the future for Beacon 2.0 work with search vs. category, I'm thinking we could rely on explicit definition of the page type via context variables OR on the explicit setting of the search controller.

**Review by korgon — CHANGES_REQUESTED** [review]
Should we keep all the performance.mark lines in?

**Comment by dkonieczek** [comment]
Removed the performance.mark markers since they were from testing. However it could be useful to add in the future throughout the code base.

**Pull request merged**